code/datums/flock/flock.dm

```diff
@@ -644,36 +643,41 @@
 		if (RL_Started) RL_UPDATE_LIGHT(T)
 
 	for(var/obj/O in T)
-		if(istype(O, /obj/machinery/door/feather))
-			// repair door
-			var/obj/machinery/door/feather/door = O
-			door.heal_damage()
-			animate_flock_convert_complete(O)
-		else
-			for(var/keyPath in flock_conversion_paths) //types are converted with priority determined by list order
-				var/obj/replacementPath = flock_conversion_paths[keyPath] //put subclasses ahead of superclasses in the flock_conversion_paths list
-				if(istype(O, keyPath))
-					if(isnull(replacementPath))
+		for(var/keyPath in flock_conversion_paths) //types are converted with priority determined by list order
+			if(!istype(O, keyPath))
+				continue
+			if (istype(O, /obj/machinery))
+				if (istype(O, /obj/machinery/door))
+					// split for readability
+					if (istype(O, /obj/machinery/door/firedoor/pyro) || istype(O, /obj/machinery/door/window) || istype(O, /obj/machinery/door/airlock/pyro/glass/windoor) || istype(O, /obj/machinery/door/unpowered/wood))
 						qdel(O)
-					else
-						var/dir = O.dir
-						var/obj/converted = new replacementPath(T)
-						// if the object is a closet, it might not have spawned its contents yet
-						// so force it to do that first
-						if(istype(O, /obj/storage))
-							var/obj/storage/S = O
-							if(!isnull(S.spawn_contents))
-								S.make_my_stuff()
-						// if the object has contents, move them over!!
-						for (var/obj/OO in O)
-							OO.set_loc(converted)
-						for (var/mob/M in O)
-							M.set_loc(converted)
+						break
+					if (istype(O, /obj/machinery/door/poddoor/pyro) && !(findtext(O.name, "cargo") || findtext(O.name, "pod") || findtext(O.name, "mass") || findtext(O.name, "disposal"))) // shutters
```

Comment:
I still really don't like relying on `findtext(name, "blah")` - it assumes that mappers have stuck to an unofficial naming scheme, and haven't made a mistake. 

I think a better way would be to check whether the turf is dense without the door, and if it is delete it - otherwise convert it.

Reply:
i'd rather just fix this outright on the upstream by making it a new type or subtype or whatever

Reply:
that seems like a much nicer fix